compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/ContextMenuProvider.desktop.kt

```diff
@@ -97,30 +98,51 @@ private val LocalContextMenuData = staticCompositionLocalOf<ContextMenuData?> {
     null
 }
 
-private fun Modifier.contextMenuDetector(
-    state: ContextMenuState,
-    enabled: Boolean = true
+
+/**
+ * Detects events that open a context menu (mouse right-clicks).
+ *
+ * @param key The pointer input handling coroutine will be cancelled and **re-started** when
+ * [contextMenuOpenDetector] is recomposed with a different [key].
+ * @param enabled Whether to enable the detection.
+ * @param isOpen Whether the context menu intended to be opened is currently open.
+ * @param onOpen Invoked when a context menu opening event is detected, with the local offset it
+ * should be opened at.
+ */
+fun Modifier.contextMenuOpenDetector(
```

Comment:
Ah, sure. Let's then mark it as experimental, so that other design system (not only material can use that). Or we can provide BasicMenu in `foundation` in the future and won't expose this api